main/test/ca/uwaterloo/flix/TestStreaming.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 Magnus Madsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix
+
+import ca.uwaterloo.flix.api.Flix
+import org.scalatest.funsuite.AnyFunSuite
+
+import java.nio.file.{Files, Paths}
+
+class TestStreaming extends AnyFunSuite with TestUtils {
+
+  /**
+    * The number of pieces each input is broken down into.
+    */
+  private val Chunks: Int = 100
+
+  test("Stream01.simple-card-game") {
+    val input = Files.readString(Paths.get("examples/simple-card-game.flix"))
+    compile(input)
+  }
+
+  test("Stream02.the-ast-typing-problem-with-polymorphic-records") {
+    val input = Files.readString(Paths.get("examples/the-ast-typing-problem-with-polymorphic-records.flix"))
+    compile(input)
+  }
+
+  test("Stream03.using-channels-and-select") {
+    val input = Files.readString(Paths.get("examples/using-channels-and-select.flix"))
+    compile(input)
+  }
+
+  /**
+    * We break the given string `input` down into a number of chunks and compile each of them.
+    *
+    * We simply test that compilation does not crash with an unexpected exception.
+    */
+  private def compile(input: String): Unit = {
+    val length = input.length
+    val step = length / Chunks
+
+    val flix = new Flix()
+    flix.compile()
+    for (i <- 0 until Chunks) {
+      val e = Math.min(i * step, length)
+      val s = input.substring(0, e)
```

Comment:
This seems to add code from the beginning over and over.

It might be simpler just to used `StringOps.grouped`?

Reply:
What's the benefit

Reply:
Oh I see. It's a helper.

Reply:
Cannot use grouped. I compute prefixes.

Reply:
Ah, didn't realize that was intentional.